WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyRingViewModel.cs

```diff
@@ -74,82 +72,61 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 						.PrivacyProgressUpdateTrigger
 						.CombineLatest(sizeTrigger)
 						.ObserveOn(RxApp.MainThreadScheduler)
-						.Subscribe(x =>
-						{
-							var usableHeight = x.Second.Item2;
-							var usableWidth = x.Second.Item1;
-							Margin = new Thickness(usableWidth / 2, usableHeight / 2, 0, 0);
-							NegativeMargin = new Thickness(Margin.Left * -1, Margin.Top * -1, 0, 0);
-							RefreshCoinsList(itemsSourceList);
-						})
+						.Subscribe(_ => RenderRing(itemsSourceList))
 						.DisposeWith(disposables);
 	}
 
-	private void RefreshCoinsList(SourceList<PrivacyRingItemViewModel> itemsSourceList)
+	private void RenderRing(SourceList<PrivacyRingItemViewModel> list)
 	{
-		var pockets = _walletViewModel.Wallet.GetPockets();
-		itemsSourceList.Edit(list => CreateSegments(pockets, list));
+		SetMargins();
+
+		Items.Clear();
+		list.Edit(list => CreateSegments(list));
+
+		PreviewItems.RemoveRange(1, PreviewItems.Count - 1);
+		PreviewItems.AddRange(list.Items);
+
+		SetReferences(list);
 	}
 
-	private void CreateSegments(IEnumerable<Pocket> pockets, IExtendedList<PrivacyRingItemViewModel> list)
+	private void CreateSegments(IExtendedList<PrivacyRingItemViewModel> list)
 	{
-		Items.Clear();
 		list.Clear();
 
 		if (Width == 0d)
 		{
 			return;
 		}
```

Comment:
Consider moving this check into `RenderRing()` to save some unnecessary calculations.